### PR TITLE
fixed #18161: Crash unplugging USB on Windows

### LINKED ISSUE #18161: Crash unplugging USB on Windows
### Issue type

Crash or freeze

### Bug description

I use a Scarlett USB on Windows for a microphone and also headphones. If I start Musescore with it plugged in (or not) all is good. If I unplug the USB when Musescore is running it causes an access violation and Musescore crashes (every time.) It does not crash if I plug in the USB when the program is running.

### Steps to reproduce

On Windows 11, run Musescore and plugin a sound device using a USB. Now, unplug the USB and crash.

### Screenshots/Screen recordings

_No response_

### MuseScore Version

Latest master (june 24) v4.2

### Regression

I don't know

### Operating system

Windows 11

### Additional context

It crashes in audiodeviceslisterner.cpp in method OnDefaultDeviceChanged() line 121:

if (m_previousIdString == new_default_device_id) {

The first string is a winrt::hstring and the second is a pointer which is NULL when you unplug a USB. The winrt::hstring == operator will blow up if "new_default_device_id" is NULL.

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #18161

### CODE REVIEW

File: src/framework/audio/internal/platform/win/audiodeviceslistener.cpp [hunk 1]
```diff
@@ -118,11 +118,11 @@ HRESULT AudioDevicesListener::OnDefaultDeviceChanged(EDataFlow flow, ERole role,
         return S_OK;
     }
 
-    if (m_previousIdString == new_default_device_id) {
+    if (m_previousDefaultDeviceId == new_default_device_id) {
```
Comment (cbjeukendrup) [inline src/framework/audio/internal/platform/win/audiodeviceslistener.cpp:121]: Now that `m_previousDefaultDeviceId` and `new_default_device_id` are both plain `LPCWSTR` (aka `wchar_t const *`), this simply compares the pointers, and does not compare the strings. Is that desired, or should maybe `lstrcmpW` be used instead?

Comment (laturetab) [reply src/framework/audio/internal/platform/win/audiodeviceslistener.cpp:121]: Microsoft docs say the pointer is only valid for the duration of the call -- so you will need to copy the string into something and do a full string compare.

Comment (Eism) [reply src/framework/audio/internal/platform/win/audiodeviceslistener.cpp]: ok, looks like initializing hstring from NULL doesn't work at all. I just added a check as recommended in the issue

**Review by cbjeukendrup — APPROVED** [review]

**Comment by zacjansheski** [comment]
I am still getting the crash on these checks

[AudioInterfaceCrash.zip](https://github.com/musescore/MuseScore/files/11958879/AudioInterfaceCrash.zip)

**Pull request merged**